connector/src/main/java/org/geysermc/connector/utils/SkinProvider.java

```diff
@@ -164,11 +164,17 @@ private static Cape supplyCape(String capeUrl) {
         BufferedImage image = ImageIO.read(new URL(imageUrl));
         Geyser.getLogger().debug("Downloaded " + imageUrl);
 
-        if (cape) {
+        if (cape && image.getWidth() <= 64) {
+            BufferedImage newImage = new BufferedImage(64, 32, BufferedImage.TYPE_INT_RGB);
+            Graphics g = newImage.createGraphics();
+            g.drawImage(image, 0, 0, image.getWidth(), image.getHeight(), null);
+            g.dispose();
+            image = newImage;
+        }
+        else if(cape && image.getWidth() >= 64 && image.getWidth() <= 128) {
             BufferedImage newImage = new BufferedImage(64, 32, BufferedImage.TYPE_INT_RGB);
-
             Graphics g = newImage.createGraphics();
-            g.drawImage(image, 0, 0, 64, 32, null);
+            g.drawImage(scale(image), 0, 0, scale(image).getWidth(), scale(image).getHeight(), null);
```

Comment:
You are scaling the image 3 times. You can fix this by making it a local variable

Reply:
Whoops! Didn't even notice that, my bad. I'll fix that up ASAP!

Reply:
And btw you can just do `image = image.getWidth() > 64 ? scale(image) : image;` after the old `if (cape)` check to remove duplicate code